hw07-patterns/src/main/java/ru/otus/edu/levina/atm/api/Cell.java

```diff
@@ -0,0 +1,61 @@
+package ru.otus.edu.levina.atm.api;
+
+import java.util.Map;
+
+/**
+ * Keeps banknotes of the same nominal
+ * 
+ * @author levina
+ *
+ */
+public interface Cell {
+    /**
+     * Chain of responsibility
+     * @param cell
+     */
+    void setNext(Cell atm);
+    Cell getNext();
+    
+    
+    /**
+     * Returns a nominal of this cell
+     * 
+     * @return
+     */
+    Nominal getNominal();
+
+    /**
+     * Returns a number of banknotes in the cell
+     * 
+     * @return
+     */
+    int getAvailableBanknoteCount();
+
+
+    /**
+     * Selects suitable banknotes from the batch
+     * 
+     * @param bns a batch of banknotes
+     * @return not suitable banknotes
+     * 
+     * @throws IllegalArgumentException
+     *             if banknote quantity is negative or null
+     */
+    Map<Banknote, Integer> process(Map<Banknote, Integer> bns);
```

Comment:
Пока не прочитал комментарий не смог понять, что означает `bns`.
`BaNknoteS` ?

Лучше так и назвать
```suggestion
    Map<Banknote, Integer> process(Map<Banknote, Integer> banknotes);
```

В прошлом ДЗ не обратил внимания.